### PR TITLE
Removes option to disable stack overflow guard

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This removes risk of compilation drift by removing the option to disable stack overflow protection.

### CODE REVIEW

File: wasm/naivevm/vm_stack.go [hunk 1]
```diff
@@ -137,12 +137,10 @@ func (s *frameStack) pop() *frame {
 }
 
 func (s *frameStack) push(val *frame) {
-	if buildoptions.CheckCallStackOverflow {
+	if s.sp+1 == len(s.stack) {
 		if buildoptions.CallStackHeightLimit <= s.sp {
```
Comment (codefromthecrypt) [inline wasm/naivevm/vm_stack.go:141]: I moved this inside so that it is only hit when actually growing

File: wasm/buildoptions/stackoverflow.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package buildoptions
+
+// CallStackHeightLimit is the maximum WebAssembly call stack height. This allows wazero to raise
```
Comment (codefromthecrypt) [inline wasm/buildoptions/stackoverflow.go:3]: if this doc is inaccurate help me make it better!

File: wasm/wazeroir/interpreter.go [hunk 3]
```diff
@@ -70,10 +70,8 @@ func (it *interpreter) drop(r *InclusiveRange) {
 }
 
 func (it *interpreter) pushFrame(frame *interpreterFrame) {
-	if buildoptions.CheckCallStackOverflow {
-		if buildoptions.CallStackHeightLimit <= len(it.frames)-1 {
-			panic(wasm.ErrCallStackOverflow)
-		}
+	if buildoptions.CallStackHeightLimit <= len(it.frames)-1 {
```
Comment (mathetake) [inline wasm/wazeroir/interpreter.go:73]: While there let's reduce the cost of each operation😄
```suggestion
	if buildoptions.CallStackHeightLimit <= len(it.frames) {
```

**Review by mathetake — APPROVED** [review]
Thanks!

**Comment by codefromthecrypt** [comment]
I owe vm_stack_test.go push test backfill prior to merge

**Review by mathetake — APPROVED** [review]
Thanks!!!!

**Comment by codefromthecrypt** [comment]
Thanks for the help!

**Pull request merged**